block/manager_test.go

```diff
@@ -282,6 +297,84 @@ func getTempKVStore(t *testing.T) ds.TxnDatastore {
 	return kvStore
 }
 
+// Test_submitBlocksToDA_BlockMarshalErrorCase1: A itself has a marshalling error. So A, B and C never get submitted.
+func Test_submitBlocksToDA_BlockMarshalErrorCase1(t *testing.T) {
```

Comment:
@arhamj I think there is some confusion. I just wanted to add the marshal error cases back into `TestSubmitBlocksToDA`. 

We shouldn't need to add any new mocking or these separate tests. 

These tests you've added are testing that there is in fact an edge case in marshalling that gets by the reading from disk. That isn't want I was implying needs to be protected against. Between reading from disk and trying to marhsal  in submit blocks in submit blocks, the data struct is handled. This handling changes over time and could potentially introduce bugs or issues in the future. That is why we want to test the marshal edge case in submit blocks. 

We can remove these two tests and just put them back into `TestSubmitBlocksToDA` as test cases. 

You can also remove the added mock code as that shouldn't be needed.

Reply:
OK. This is a tough one.
We can't use normal store because it can't handle objects that can't be serialized / deserialized.
There is no place in the code where we can "invalidate" header retrieved from normal store.

I see two options here:
 * use mock store as proposed
 * refactor `submitBlocksToDA` to receive list of blocks instead of reading it from store; caller should read blocks from store and pass to `submitBlocksToDA` (or in case of test, we can just provide test-specific blocks)

Reply:
OK, let's leave as is then.